lib/puppet/provider/agent_upgrade/windows.rb

```diff
@@ -0,0 +1,89 @@
+Puppet::Type.type(:agent_upgrade).provide(:agent_upgrade) do
+
+  @doc = <<-EOT
+    Windows provider to upgrade from 3.8 to 4.0 using latest puppet-agent version,
+    requires PowerShell to be already installed, only upgrade if < 4.0
+  EOT
+  confine :osfamily => /windows/i
```

Comment:
If there is only one and we are confine to windows and any other provider, which I don't see happening confines to there OS there is no need for default as they are all confined